addons/l10n_latam_check/models/account_chart_template.py

```diff
@@ -0,0 +1,41 @@
+from odoo import models, Command, api, _
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class AccountChartTemplate(models.Model):
+    _inherit = 'account.chart.template'
+
+    @api.model
+    def _get_third_party_checks_country_codes(self):
+        """ Return the list of country codes for the countries where third party checks journals should be created
+        when installing the COA"""
+        return ["AR"]
+
+    def _create_bank_journals(self, company, acc_template_ref):
+        res = super()._create_bank_journals(company, acc_template_ref)
+
+        if company.country_id.code in self._get_third_party_checks_country_codes():
+            self.env['account.journal'].create([{
+                'name': _('Third Party Checks'),
+                'type': 'cash',
+                'company_id': company.id,
+                'outbound_payment_method_line_ids': [
+                    Command.create({'payment_method_id': self.env.ref('l10n_latam_check.account_payment_method_out_third_party_checks').id}),
+                ],
+                'inbound_payment_method_line_ids': [
+                    Command.create({'payment_method_id': self.env.ref('l10n_latam_check.account_payment_method_new_third_party_checks').id}),
+                    Command.create({'payment_method_id': self.env.ref('l10n_latam_check.account_payment_method_in_third_party_checks').id}),
+                ]}])
+            self.env['account.journal'].create([{
```

Comment:
Hi @william-andre 

I have updated the code to make it two separate creates without using a list so this part of the code is resolved. This was dirty code because formerly we were using the create val_list approach.

NOTE: In case it adds up, I just want to let you know that the reason we made this change is because we found a problem when creating the journals in the create with val_list:  is not properly computing the code to be used in the new journal, use the same code for both journals. It creates the first journal with code CSH2, and the second journal with the same CSH2 and returns an error. Leave you here the log:

2022-11-23 23:27:52,534 62 ERROR 16_checks_2 odoo.sql_db: bad query: INSERT INTO "account_journal" ("active", "bank_statements_source", "check_manual_sequencing", "code", "color", "company_id", "create_date", "create_uid", "default_account_id", "invoice_reference_model", "invoice_reference_type", "loss_account_id", "name", "payment_sequence", "profit_account_id", "refund_sequence", "sale_activity_type_id", "sequence", "show_on_dashboard", "type", "write_date", "write_uid") VALUES (true, 'undefined', false, 'CSH2', 0, 82, '2022-11-23 23:27:48.813476', 34, 4719, 'odoo', 'invoice', 4708, 'Third Party Checks', true, 4699, false, NULL, 10, true, 'cash', '2022-11-23 23:27:48.813476', 34), (true, 'undefined', false, 'CSH2', 0, 82, '2022-11-23 23:27:48.813476', 34, 4720, 'odoo', 'invoice', 4708, 'Rejected Third Party Checks', true, 4699, false, NULL, 10, true, 'cash', '2022-11-23 23:27:48.813476', 34) RETURNING "id"
```suggestion
                ]}])
            self.env['account.journal'].create([{
```
ERROR: duplicate key value violates unique constraint "account_journal_code_company_uniq"
DETAIL:  Key (company_id, code)=(82, CSH2) already exists.
 
2022-11-23 23:27:52,540 62 INFO 16_checks_2 unittest.suite: ====================================================================== 
2022-11-23 23:27:52,540 62 ERROR 16_checks_2 unittest.suite: ERROR: setUpClass (odoo.addons.l10n_latam_check.tests.test_own_checks.TestOwnChecks)
Traceback (most recent call last):
  File "/home/odoo/custom/odoo/addons/l10n_latam_check/tests/common.py", line 16, in setUpClass
    cls.company_data_3 = cls.setup_company_data(
  File "/home/odoo/custom/odoo/addons/account/tests/common.py", line 212, in setup_company_data
    chart_template.try_loading(company=company, install_demo=False)
  File "/home/odoo/custom/odoo/addons/account/models/chart_template.py", line 249, in try_loading
    template.with_context(default_company_id=company.id)._load(company)
  File "/home/odoo/custom/odoo/addons/l10n_ar/models/account_chart_template.py", line 66, in _load
    res = super()._load(company)
  File "/home/odoo/custom/enterprise/account_followup/models/chart_template.py", line 10, in _load
    res = super(AccountChartTemplate, self)._load(company)
  File "/home/odoo/custom/enterprise/account_reports/models/chart_template.py", line 9, in _load
    res = super(AccountChartTemplate, self)._load(company)
  File "/home/odoo/custom/odoo/addons/account/models/chart_template.py", line 394, in _load
    self._create_bank_journals(company, acc_template_ref)
  File "/home/odoo/custom/odoo/addons/l10n_latam_check/models/account_chart_template.py", line 21, in _create_bank_journals
    self.env['account.journal'].create([{
  File "<decorator-gen-212>", line 2, in create
  File "/home/odoo/custom/odoo/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/odoo/custom/odoo/addons/account_check_printing/models/account_journal.py", line 62, in create
    journals = super().create(vals_list)
  File "<decorator-gen-200>", line 2, in create
  File "/home/odoo/custom/odoo/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/odoo/custom/odoo/addons/account/models/account_journal.py", line 648, in create
    journals = super(AccountJournal, self.with_context(mail_create_nolog=True)).create(vals_list)
  File "<decorator-gen-142>", line 2, in create
  File "/home/odoo/custom/odoo/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/odoo/custom/odoo/addons/mail/models/mail_thread.py", line 258, in create
    threads = super(MailThread, self).create(vals_list)
  File "<decorator-gen-65>", line 2, in create
  File "/home/odoo/custom/odoo/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/odoo/custom/odoo/odoo/addons/base/models/ir_fields.py", line 665, in create
    recs = super().create(vals_list)
  File "<decorator-gen-13>", line 2, in create
  File "/home/odoo/custom/odoo/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/odoo/custom/odoo/odoo/models.py", line 3925, in create
    records = self._create(data_list)
  File "/home/odoo/custom/odoo/odoo/models.py", line 4106, in _create
    cr.execute(
  File "/home/odoo/custom/odoo/odoo/sql_db.py", line 315, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "account_journal_code_company_uniq"
DETAIL:  Key (company_id, code)=(82, CSH2) already exists.